src/ngMessages/messages.js

```diff
@@ -24,45 +24,64 @@ var jqLite = angular.element;
- * template directives.
+ * The `ngMessages` directive allows individual keys in a key/value collection to each be associated with an element
+ * (or 'message') that will show or hide based on the truthiness of that key's value in the collection. A common use
+ * case for `ngMessages` is for annotating form inputs with error messages using the `$error` object exposed by the
+ * {@link ngModel ngModel} directive.
+ *
+ * Setting an element's `ngMessages` attribute to the name of a key/value collection will allow its child elements to
+ * be conditionally shown or hidden based on whether their associated keys (declared on each using `ngMessage`)
+ * have truthy values.
+ *
+ *
+ * Consider the following example, which illustrates a typical use case of `ngMessages`. Within the form `myForm` we
+ * have a text input named `myField` which is bound to the scope variable `field` using the {@link ngModel ngModel}
+ * directive.
+ *
+ * The `myField` field is a required field with a maximum length of 15 characters.
  *
  * ```html
  * <form name="myForm">
  *   <label>
  *     Enter text:
- *     <input type="text" ng-model="field" name="myField" required minlength="5" />
+ *     <input type="email" ng-model="field" name="myField" required maxlength="15" />
  *   </label>
  *   <div ng-messages="myForm.myField.$error" role="alert">
- *     <div ng-message="required">You did not enter a field</div>
- *     <div ng-message="minlength, maxlength">
- *       Your email must be between 5 and 100 characters long
- *     </div>
+ *     <div ng-message="required">Please enter a value for this field.</div>
+ *     <div ng-message="email">This field must be a valid email address.</div>
+ *     <div ng-message="maxlength">This field can be at most 15 characters long.</div>
  *   </div>
  * </form>
  * ```
  *
- * Now whatever key/value entries are present within the provided object (in this case `$error`) then
- * the ngMessages directive will render the inner first ngMessage directive (depending if the key values
- * match the attribute value present on each ngMessage directive). In other words, if your errors
- * object contains the following data:
+ * In order to show error messages corresponding to `myField` we first create an element with an `ngMessages` attribute
+ * set to the `$error` object owned by the `myField` input in our `myForm` form.
+ *
+ * Within this element we then create separate elements for each of the possible errors that the user's submission for
+ * `myField` could have. The `ngMessage` attribute is used to declare which element(s) will appear for which error
+ * - for example, setting `ng-message="required"` specifies that this particular element should be displayed when there
+ * is no value present for the required field `myField` (because the key `required` will be `true` in the object
+ * `myForm.myField.$error`).
+ *
+ * By default, `ngMessages` will only display one message for a particular key/value collection at any time. If more
```

Comment:
"If more than one message is applicable then the decision of which message to show is" -> "If more than one message (or error) key is currently true, then which element is shown is"